benchmarks/src/charconv_floats.cpp

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <benchmark/benchmark.h>
+#include <charconv>
+#include <random>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <system_error>
+#include <type_traits>
+#include <vector>
+
+using namespace std;
+
+void verify(const bool b) {
+    if (!b) {
+        puts("FAIL");
+        exit(EXIT_FAILURE);
+    }
+}
+
+enum class RoundTrip { Sci, Fix, Gen, Hex, Lossy };
+
+constexpr chars_format chars_format_from_RoundTrip(const RoundTrip rt) {

```

Comment:
This could be upgraded to `consteval` which would detect misuse at compile-time.